core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/DatasourceService.java

```diff
@@ -154,4 +170,54 @@ public static boolean isDSWDatasource( Domain domain ) {
     }
     return !isMetadataDatasource( domain );
   }
+
+  public boolean isDSWDatasource( String domainId ) {
+    return !isMetadataDatasource( domainId );
+  }
+
+  protected int getDatasourceLoadThreadCount() throws IllegalArgumentException {
```

Comment:
No direction, but look out for further efforts.

Don't add to the method signature `throws IllegalArgumentException` since IllegalArgumentException is a subclass of  RuntimeException: https://wiki.sei.cmu.edu/confluence/display/java/ERR07-J.+Do+not+throw+RuntimeException%2C+Exception%2C+or+Throwable

https://docs.oracle.com/javase/7/docs/api/java/lang/IllegalArgumentException.html

You can see in core java classes, they don't throw runtime Exceptions:
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Executors.html#newFixedThreadPool-int-
https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html#parseInt-java.lang.String-

Reply:
agreed. I didn't say anything about adding what runtime exceptions the method throws in the java documentation section. Just follow what the oracle java documentation does for runtime exceptions.

Reply:
there was some confusion on my part.Ignore my comment. I apologize.